third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -295,4 +295,14 @@ impl<'r> TransactionDataCache<'r> {
         let ty_layout = loader.type_to_type_layout(&ty)?;
         Ok(self.event_data.push((guid, seq_num, ty, ty_layout, val)))
     }
+
+    pub(crate) fn emitted_events(&self, guid: Vec<u8>, ty: Type) -> PartialVMResult<Vec<Value>> {
+        let mut events = vec![];
+        for event in self.event_data.iter() {
+            if event.0 == guid && event.2 == ty {
+                events.push(event.4.copy_value()?);
+            }
+        }
+        Ok(events)
+    }
```

Comment:
No need to do it in this PR, but we really should move the event buffer out of the Move VM and re implement it as a native extension. Events being part of the Move VM is a legacy of the past we haven't got the time to clean up.

Reply:
I don't mind if DW can finish that in this PR 💯